fe2-android/app/src/androidTest/java/com/github/dedis/popstellar/ui/detail/event/consensus/ElectionStartFragmentTest.java

```diff
@@ -67,36 +65,42 @@
 import java.util.Collections;
 import java.util.concurrent.TimeUnit;
 
+import javax.inject.Inject;
+
+import dagger.hilt.android.testing.HiltAndroidRule;
+import dagger.hilt.android.testing.HiltAndroidTest;
 import io.reactivex.Observable;
 
-@RunWith(AndroidJUnit4ClassRunner.class)
+@HiltAndroidTest
 public class ElectionStartFragmentTest {
 
+  @Inject AndroidKeysetManager keysetManager;
+  @Inject Gson gson;
+
   // A custom rule to call setup and teardown before the fragment rule and after the mockito rule
   private final TestRule setupRule =
       new ExternalResource() {
         @Override
         protected void before() {
+          // Injection with hilt
+          hiltRule.inject();
           // Preload the data schema before the test run
           JsonUtils.loadSchema(JsonUtils.DATA_SCHEMA);
 
           when(remoteDataSource.incrementAndGetRequestId()).thenReturn(42);
           when(remoteDataSource.observeWebsocket()).thenReturn(Observable.empty());
-          Observable<GenericMessage> upstream =
-              Observable.fromArray((GenericMessage) new Result(42));
+          Observable<GenericMessage> upstream = Observable.fromArray(new Result(42));
 
           // Mock the remote data source to receive a response
           when(remoteDataSource.observeMessage()).thenReturn(upstream);
 
           try {
-            LAORepository.destroyInstance();
             laoRepository =
-                LAORepository.getInstance(
+                new LAORepository(
                     remoteDataSource,
                     localDataSource,
-                    Injection.provideAndroidKeysetManager(
-                        ApplicationProvider.getApplicationContext()),
-                    Injection.provideGson(),
+                    keysetManager,
+                    gson,
                     new ProdSchedulerProvider());
```

Comment:
The test fails because the LAORepository of the ElectionStartFragment is no longer replaced by this one before the test.
Is it possible to uninstall/reinject this laoRepository inside the ElectionStartFragment before the test run ?

Reply:
It was working before I made some fixes...
I will have to inject LAORepository manually for the test.

Reply:
Fixed